contracts/modules/cover/CoverOrder.sol

```diff
@@ -0,0 +1,225 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import "@openzeppelin/contracts-v4/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts-v4/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-v4/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts-v4/access/Ownable.sol";
+
+import "../../interfaces/ICoverOrder.sol";
+import "../../interfaces/IMemberRoles.sol";
+import "../../interfaces/INXMMaster.sol";
+import "../../interfaces/INXMToken.sol";
+import "../../interfaces/IWeth.sol";
+import "../../interfaces/IPool.sol";
+
+contract CoverOrder is ICoverOrder, Ownable {
+  using ECDSA for bytes32;
+  using SafeERC20 for IERC20;
+
+  /* ========== STATE VARIABLES ========== */
+  mapping(bytes32 => bool) public executedOrders;
+
+  /* ========== IMMUTABLES ========== */
+  ICover public immutable cover;
+  IMemberRoles public immutable memberRoles;
+  INXMToken public immutable nxmToken;
+  INXMMaster public immutable master;
+  IWeth public immutable weth;
+
+  /* ========== CONSTANTS ========== */
+  address public constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+  uint private constant ETH_ASSET_ID = 0;
+  uint private constant NXM_ASSET_ID = type(uint8).max;
+
+  /* ========== CONSTRUCTOR ========== */
+  constructor(
+    address _cover,
+    address _memberRoles,
+    address _nxmToken,
+    address _master,
+    address _weth,
+    address _owner
+  ) {
+    master = INXMMaster(_master);
+    cover = ICover(_cover);
+    memberRoles = IMemberRoles(_memberRoles);
+    nxmToken = INXMToken(_nxmToken);
```

Comment:
`Cover`, `MemberRoles` and `nxmToken` can be pulled from `Master` to simplify the params and have less room for error.